message_ix_models/tools/costs/projections.py

```diff
@@ -458,8 +468,21 @@ def create_message_outputs(
         )
         .astype(dtypes)
         .query("year_vtg in @config.Y")
+        .assign(first_technology_year=lambda x: x.first_technology_year.astype(float))
+        .assign(first_technology_year=lambda x: x.first_technology_year.astype(int))
+        .query("year_vtg >= first_technology_year")
         .reset_index(drop=True)
-        .drop_duplicates()
+        .drop_duplicates()[
+            [
```

Comment:
AFAICT the intent of this selection is to drop the added column "first_technology_year". If so, then:
```
.drop("first_technology_year", axis=1)
```
…is more explicit and less noisy. Recall we made several adjustments like this in #99.